app/components/Dashboard/BestNewMusicTab/BestNewMusicMenu/BestNewMusicCard/index.js

```diff
@@ -1,39 +1,75 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
-import { bestNewItemShape } from '..';
+import TrackPopupContainer from '../../../../../containers/TrackPopupContainer';
 
 import styles from './styles.scss';
 
 const BestNewMusicCard = props => {
   const {
     item,
-    onClick
+    onClick,
+    withPopupMenu
   } = props;
-  
-  return (
+
+  const card = () => (
     <div
       className={styles.best_new_music_card}
-      onClick={ onClick }
+      onClick={onClick}
     >
       <div className={styles.card_thumbnail}>
-        <img alt={item.title} src={item.thumbnail}/>
+        <img alt={item.title} src={item.thumbnail} />
       </div>
       <div className={styles.card_info}>
         <div className={styles.card_title}>
-          { item.title }
+          {item.title}
         </div>
         <div className={styles.card_artist}>
-          { item.artist }
+          {item.artist}
         </div>
       </div>
     </div>
   );
+
+  const track = {
+    artist: {
+      name: item.artist
+    },
+    name: item.title
+  };
+
+  return withPopupMenu ? (
+    <TrackPopupContainer
+      trigger={card()}
+      track={track}
+      artist={item.artist}
+      title={item.title}
+      thumb={item.thumbnail}
+    />
+  ) : card();
 };
 
+export const bestNewItemShape = PropTypes.shape({
```

Comment:
Moved the base shape due to errors occurring when the shape was defined in a parent component. The shape was used before it was initialized, now everything seems to be fine.

Reply:
Nice, I couldn't figure out how to do it.